backend/app/user_handler.go

```diff
@@ -641,7 +641,32 @@ func (h *Handler) ChargeBalance(c *gin.Context) {
 		return
 	}
 
-	err = internal.TransferTFTs(h.substrateClient, request.Amount, user.Mnemonic, h.config.SystemAccount.Mnemonic)
+	requestedTFTs, err := internal.FromUSDToTFT(h.substrateClient, float64(request.Amount))
+	if err != nil {
+		log.Error().Err(err).Send()
+		InternalServerError(c)
+		return
+	}
+
+	systemUSDBalance, err := internal.GetUserBalanceUSD(h.substrateClient, h.config.SystemAccount.Mnemonic)
+	if err != nil {
+		log.Error().Err(err).Send()
+		InternalServerError(c)
+		return
+	}
+
+	if systemUSDBalance < float64(request.Amount) {
+		if err = h.db.CreatePendingRecord(&models.PendingRecord{
+			UserID:    userID,
+			TFTAmount: requestedTFTs,
+		}); err != nil {
+			log.Error().Err(err).Send()
+			InternalServerError(c)
+			return
+		}
+	}
+
+	err = internal.TransferTFTs(h.substrateClient, requestedTFTs, user.Mnemonic, h.config.SystemAccount.Mnemonic)
```

Comment:
I know we asked that we fund with whatever is available at the moment, but when I read this, it won't work, given the balance isn't guaranteed to stay the same if there're (one or more users doing the similar requests at the same time) 

I suggest this we create a pending record and the transfers to happen only in the background (by a single worker)

Reply:
according to our discussion we won't send partial balance to the user , will be removed